internal/executor/service/retry_cache.go

```diff
@@ -0,0 +1,39 @@
+package service
+
+import "sync"
+
+type RetryCache interface {
+	AddRetryAttempt(jobId string)
+	GetNumberOfRetryAttempts(jobId string) int
+	Evict(jobId string)
+}
+
+type InMemoryRetryCache struct {
+	mutex *sync.RWMutex
+	cache map[string]int
+}
+
+func NewInMemoryRetryCache() RetryCache {
```

Comment:
It would be good if this (and the other) exported function(s) could have a comment explaining what the function actually does.